### PR TITLE
scheduler: add opinfluence for balance leader scheduler

### LINKED ISSUE #2571: scheduler: no opinfluence for transferLeaderOut
## Bug Report

When we pick store for transfer leader in `transferLeaderOut`, there is no `opinfluence`.

Suppose：
There is a leader in store1 and followers in store2 and store3. And the score of store2 and store3 is the same.
When there has been some leader transfered to store2, we still pick it without `opinfluence`.

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

We miss op influence when sort target store in balance leader scheduler. To fix #2571 

### What is changed and how it works?

add it

### Check List

Tests

- Unit test

### Release note
- No release note

### CODE REVIEW

File: server/schedulers/balance_test.go [hunk 1]
```diff
@@ -304,6 +305,34 @@ func (s *testBalanceLeaderSchedulerSuite) TestScheduleWithOpInfluence(c *C) {
...
+	s.tc.AddLeaderStore(1, 7)
+	s.tc.AddLeaderStore(2, 8)
+	s.tc.AddLeaderStore(3, 9)
+	s.tc.AddLeaderStore(4, 12)
+	s.opt.TolerantSizeRatio = 0.1
+	for i := uint64(1); i <= 7; i++ {
+		s.tc.AddLeaderRegion(i, 4, 1, 2, 3)
+	}
+
+	// balance leader: 4->1, 4->1, 4->2
```
Comment (disksing) [inline server/schedulers/balance_test.go:320]: can we check if the target ids are correct?

Comment (lhy1024) [reply server/schedulers/balance_test.go:321]: There is no function to get target store id in `op`?

Comment (disksing) [reply server/schedulers/balance_test.go:321]: `op.Step(0).ToStore`

Comment (lhy1024) [reply server/schedulers/balance_test.go:319]: Thanks

**Pull request merged**